aptos-move/framework/aptos-framework/sources/account_abstraction/permissioned_delegation.move

```diff
@@ -0,0 +1,190 @@
+module aptos_framework::permissioned_delegation {
+    use std::error;
+    use std::option::Option;
+    use std::signer;
+    use aptos_std::ed25519;
+    use aptos_std::ed25519::{new_signature_from_bytes, new_unvalidated_public_key_from_bytes, UnvalidatedPublicKey};
+    use aptos_std::table::{Self, Table};
+    use aptos_framework::auth_data;
+    use aptos_framework::auth_data::AbstractionAuthData;
+    use aptos_framework::bcs_stream;
+    use aptos_framework::bcs_stream::deserialize_u8;
+    use aptos_framework::permissioned_signer::{Self, is_permissioned_signer, StorablePermissionedHandle};
+    use aptos_framework::token_bucket;
+    #[test_only]
+    use std::bcs;
+    #[test_only]
+    use std::option;
+
+    const ENOT_MASTER_SIGNER: u64 = 1;
+    const EINVALID_PUBLIC_KEY: u64 = 2;
+    const EPUBLIC_KEY_NOT_FOUND: u64 = 3;
+    const EINVALID_SIGNATURE: u64 = 4;
+    const EHANDLE_EXISTENCE: u64 = 5;
+    const ERATE_LIMITED: u64 = 6;
+
+    enum HandleBundle has store {
+        V1 { handle: StorablePermissionedHandle, bucket: Option<token_bucket::Bucket> }
+    }
+
+    struct Delegation has key {
+        handle_bundles: Table<ed25519::UnvalidatedPublicKey, HandleBundle>
+    }
+
+    inline fun fetch_handle(bundle: &mut HandleBundle, check_rate_limit: bool): &StorablePermissionedHandle {
+        let token_bucket = &mut bundle.bucket;
+        if (check_rate_limit && token_bucket.is_some()) {
+            assert!(token_bucket::request(token_bucket.borrow_mut(), 1), std::error::permission_denied(ERATE_LIMITED));
+        };
+        &bundle.handle
+    }
+
+    public fun add_permissioned_handle(
+        master: &signer,
+        key: vector<u8>,
+        max_txn_per_minute: Option<u64>,
+        expiration_time: u64,
+    ): signer acquires Delegation {
+        assert!(!is_permissioned_signer(master), error::permission_denied(ENOT_MASTER_SIGNER));
+        let addr = signer::address_of(master);
+        let pubkey = ed25519::new_unvalidated_public_key_from_bytes(key);
```

Comment:
do we want this to be tied to ed25519, or potentially delegated to another account - whose authentication we then use instead?